### DISCUSSION TITLE
cascade='all, delete-orphan' and memory consumption

### DISCUSSION THREAD

**Discussion by silenius** [sqlalchemy/sqlalchemy#10368] [Usage Questions]
Hello,

I'm using joinedload inheritance and cascade='all, delete-orphan' for my relationships, it works well .. but as the database grows the memory consumed increases linearly to the point that it takes almost 1GB now when I'm deleting an object with large collections. I know about passive_deletes but for some reasons I can't use ON DELETE / UPDATE CASCADE on the PostgreSQL side for the moment (shared database). I was wondering about alternatives..? Is there a way to do that in batch or issue DELETE statements directly rather than loading the whole collection and issuing a DELETE statement for each primary key?

Thanks!

**Comment by CaselIT** [comment]
Hi,

You can probably refer to this page in the documentation https://docs.sqlalchemy.org/en/20/orm/large_collections.html
